### PR TITLE
Add: Web Socket API invocation test with tracing enabled

### PR DESCRIPTION
**Pull request by BLasan**
### Purpose
$subject

### Issue
- https://github.com/wso2/api-manager/issues/1550

### PRs
- https://github.com/wso2/carbon-apimgt/pull/11936

### CODE REVIEW

**Review by Arshardh — DISMISSED** [review]

**Comment by BLasan** [comment]
```
2023-04-05T04:01:08.8934651Z [ERROR] org.wso2.am.integration.tests.websocket.WebSocketAPIInvocationWithTracingTestCase.setEnvironment  Time elapsed: 374.884 s  <<< FAILURE!
2023-04-05T04:01:08.8935707Z org.wso2.am.integration.test.utils.APIManagerIntegrationTestException: Error while changing server configuration
2023-04-05T04:01:08.8936923Z 	at org.wso2.am.integration.tests.websocket.WebSocketAPIInvocationWithTracingTestCase.setEnvironment(WebSocketAPIInvocationWithTracingTestCase.java:131)
2023-04-05T04:01:08.8938780Z Caused by: java.io.FileNotFoundException: /home/runner/work/product-apim/product-apim/modules/integration/tests-integration/tests-backend/src/test/resources/artifacts/AM/configFiles/webSocketWithTracing/deployment.toml (No such file or directory)
2023-04-05T04:01:08.8940153Z 	at org.wso2.am.integration.tests.websocket.WebSocketAPIInvocationWithTracingTestCase.setEnvironment(WebSocketAPIInvocationWithTracingTestCase.java:113)
```

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/websocket/WebSocketAPIInvocationWithTracingTestCase.java [hunk 1]
```diff
@@ -0,0 +1,321 @@
+package org.wso2.am.integration.tests.websocket;
```
Comment (AnuGayan) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/websocket/WebSocketAPIInvocationWithTracingTestCase.java:1]: Add the license header

**Comment by BLasan** [comment]
```
[INFO] Tests run: 9, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 1,742.547 s - in TestSuite
[INFO] 
[INFO] Results:
[INFO] 
[INFO] Tests run: 9, Failures: 0, Errors: 0, Skipped: 0
[INFO] 
[INFO] 
[INFO] --- maven-jar-plugin:2.4:jar (default-jar) @ org.wso2.carbon.am.integration.backend.test ---
[INFO] Building jar: /Users/benura/Desktop/Repositories/Public/product-apim/modules/integration/tests-integration/tests-backend/target/org.wso2.carbon.am.integration.backend.test-4.2.0-SNAPSHOT.jar
[INFO] 
[INFO] --- maven-jar-plugin:2.4:test-jar (default) @ org.wso2.carbon.am.integration.backend.test ---
[INFO] Building jar: /Users/benura/Desktop/Repositories/Public/product-apim/modules/integration/tests-integration/tests-backend/target/org.wso2.carbon.am.integration.backend.test-4.2.0-SNAPSHOT-tests.jar
[INFO] 
[INFO] --- maven-install-plugin:2.5.2:install (default-install) @ org.wso2.carbon.am.integration.backend.test ---
[INFO] Installing /Users/benura/Desktop/Repositories/Public/product-apim/modules/integration/tests-integration/tests-backend/target/org.wso2.carbon.am.integration.backend.test-4.2.0-SNAPSHOT.jar to /Users/benura/.m2/repository/org/wso2/am/org.wso2.carbon.am.integration.backend.test/4.2.0-SNAPSHOT/org.wso2.carbon.am.integration.backend.test-4.2.0-SNAPSHOT.jar
[INFO] Installing /Users/benura/Desktop/Repositories/Public/product-apim/modules/integration/tests-integration/tests-backend/pom.xml to /Users/benura/.m2/repository/org/wso2/am/org.wso2.carbon.am.integration.backend.test/4.2.0-SNAPSHOT/org.wso2.carbon.am.integration.backend.test-4.2.0-SNAPSHOT.pom
[INFO] Installing /Users/benura/Desktop/Repositories/Public/product-apim/modules/integration/tests-integration/tests-backend/target/org.wso2.carbon.am.integration.backend.test-4.2.0-SNAPSHOT-tests.jar to /Users/benura/.m2/repository/org/wso2/am/org.wso2.carbon.am.integration.backend.test/4.2.0-SNAPSHOT/org.wso2.carbon.am.integration.backend.test-4.2.0-SNAPSHOT-tests.jar
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 29:31 min
[INFO] Finished at: 2023-06-14T21:54:33+05:30
[INFO] ------------------------------------------------------------------------
```

Build is passed

**Comment by BLasan** [comment]
Some irrelevant tests are failing

``` 
2023-06-15T07:37:28.6545826Z [ERROR] Tests run: 53, Failures: 6, Errors: 0, Skipped: 5, Time elapsed: 2,872.498 s <<< FAILURE! - in TestSuite
2023-06-15T07:37:28.6552194Z [ERROR] org.wso2.am.integration.tests.server.restart.JWTRevocationServerRestartTestCase.testJWTTokenRevocation  Time elapsed: 60.665 s  <<< FAILURE!
2023-06-15T07:37:28.6559550Z java.lang.AssertionError: Response code mismatched when invoke api before Retire expected [200] but found [404]
2023-06-15T07:37:28.6561306Z 	at org.wso2.am.integration.tests.server.restart.JWTRevocationServerRestartTestCase.testJWTTokenRevocation(JWTRevocationServerRestartTestCase.java:86)
2023-06-15T07:37:28.6562382Z 
2023-06-15T07:37:28.6563764Z [ERROR] org.wso2.am.integration.tests.server.restart.GraphqlServerRestartTestCase.testInvokeGraphqlAPIUsingJWTApplication  Time elapsed: 7.392 s  <<< FAILURE!
2023-06-15T07:37:28.6565161Z java.lang.AssertionError: Response code is not as expected expected [200] but found [404]
2023-06-15T07:37:28.6566695Z 	at org.wso2.am.integration.tests.server.restart.GraphqlServerRestartTestCase.testInvokeGraphqlAPIUsingJWTApplication(GraphqlServerRestartTestCase.java:180)
2023-06-15T07:37:28.6567612Z 
2023-06-15T07:37:28.6568909Z [ERROR] org.wso2.am.integration.tests.server.restart.GraphqlServerRestartTestCase.testInvokeGraphqlAPIUsingOAuthApplication  Time elapsed: 6.82 s  <<< FAILURE!
2023-06-15T07:37:28.6570146Z java.lang.AssertionError: Response code is not as expected expected [200] but found [404]
2023-06-15T07:37:28.6571661Z 	at org.wso2.am.integration.tests.server.restart.GraphqlServerRestartTestCase.testInvokeGraphqlAPIUsingOAuthApplication(GraphqlServerRestartTestCase.java:207)
2023-06-15T07:37:28.6572581Z 
2023-06-15T07:37:28.6574316Z [ERROR] org.wso2.am.integration.tests.server.restart.DefaultVersionAPIServerRestartTestCase.createNewVersionWithDefaultVersionOptionAndVerifyDefaultAPIBreaking  Time elapsed: 61.383 s  <<< FAILURE!
2023-06-15T07:37:28.6576383Z java.lang.AssertionError: Default version API test failed while invoking the API. expected [<response><value>Received Request</value></response>] but found [{"code":"404","type":"Status report","message":"Not Found","description":"The requested resource is not available."}]
2023-06-15T07:37:28.6578520Z 	at org.wso2.am.integration.tests.server.restart.DefaultVersionAPIServerRestartTestCase.createNewVersionWithDefaultVersionOptionAndVerifyDefaultAPIBreaking(DefaultVersionAPIServerRestartTestCase.java:88)
2023-06-15T07:37:28.6579968Z 
2023-06-15T07:37:28.6581623Z [ERROR] org.wso2.am.integration.tests.server.restart.AccessibilityOfBlockAPIServerRestartTestCase.testInvokeAPIBeforeChangeAPILifecycleToBlock  Time elapsed: 61.604 s  <<< FAILURE!
2023-06-15T07:37:28.6583548Z java.lang.AssertionError: Response code mismatched when invoke api before block expected [200] but found [404]
2023-06-15T07:37:28.6584820Z 	at org.wso2.am.integration.tests.server.restart.AccessibilityOfBlockAPIServerRestartTestCase.testInvokeAPIBeforeChangeAPILifecycleToBlock(AccessibilityOfBlockAPIServerRestartTestCase.java:79)
2023-06-15T07:37:28.6585793Z 
2023-06-15T07:37:28.6586692Z [ERROR] org.wso2.am.integration.tests.server.restart.APILoggingServerRestartTest.testAPIPerAPILoggingTestcase  Time elapsed: 61.308 s  <<< FAILURE!
2023-06-15T07:37:28.6587651Z java.lang.AssertionError: Invocation fails for GET request expected [200] but found [404]
2023-06-15T07:37:28.6588629Z 	at org.wso2.am.integration.tests.server.restart.APILoggingServerRestartTest.testAPIPerAPILoggingTestcase(APILoggingServerRestartTest.java:71)```

**Review by senthuran16 — APPROVED** [review]

**Comment by BLasan** [comment]
Build is passing. Merging the PR

**Pull request merged**